lib/trino-parquet/src/main/java/io/trino/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -112,10 +116,35 @@ public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor
             case BINARY:
                 return createDecimalColumnReader(descriptor).orElse(new BinaryColumnReader(descriptor));
             case FIXED_LEN_BYTE_ARRAY:
-                return createDecimalColumnReader(descriptor)
-                        .orElseThrow(() -> new TrinoException(NOT_SUPPORTED, " type FIXED_LEN_BYTE_ARRAY supported as DECIMAL; got " + descriptor.getPrimitiveType().getOriginalType()));
+                Optional<PrimitiveColumnReader> decimalColumnReader = createDecimalColumnReader(descriptor);
+                if (decimalColumnReader.isPresent()) {
+                    return decimalColumnReader.get();
+                }
+                if (isLogicalUuid(descriptor.getPrimitiveType())) {
+                    return new UuidColumnReader(descriptor);
+                }
+                if (descriptor.getPrimitiveType().getLogicalTypeAnnotation() == null) {
+                    // Iceberg 0.11.1 writes UUID as FIXED_LEN_BYTE_ARRAY without logical type annotation (see https://github.com/apache/iceberg/pull/2913)
```

Comment:
Can we access the version of Iceberg which wrote the file. Can we we limit this code path only to specific Iceberg versions?

Reply:
this i don't know. @phd3 do you happen to know?

Reply:
I think it's possible to get spec-version (1 or 2), but not library version. (based on when I looked into this [last](https://the-asf.slack.com/archives/CF01LKV9S/p1607135577408900)) I do not know if anything changed recently.